internal/utils/telegraf/emitter/baseemitter/baseemitter.go

```diff
@@ -0,0 +1,224 @@
+package baseemitter
+
+import (
+	"sync"
+	"time"
+
+	"github.com/signalfx/golib/datapoint"
+	"github.com/signalfx/golib/event"
+	"github.com/signalfx/signalfx-agent/internal/monitors/types"
+	"github.com/signalfx/signalfx-agent/internal/utils"
+	"github.com/signalfx/telegraf/plugins/outputs/signalfx/parse"
+	log "github.com/sirupsen/logrus"
+)
+
+// GetTime returns the first timestamp from an array of timestamps
+func GetTime(t ...time.Time) time.Time {
+	if len(t) > 0 {
+		return t[0]
+	}
+	return time.Now()
+}
+
+// BaseEmitter immediately converts a telegraf measurement into datapoints and
+// sends them through Output
+type BaseEmitter struct {
+	lock   *sync.Mutex
```

Comment:
Use of pointer here is unnecessary.  Default value of sync.Mutex is an unlocked mutex.